completion_aggregator/api/v1/views.py

```diff
@@ -388,3 +390,109 @@ def get(self, request, course_key):
             many=True
         )
         return paginator.get_paginated_response(serializer.data)
+
+
+class CourseLevelCompletionView(CompletionViewMixin, APIView):
+    """
+    API view to render stats for a single course.
+
+    **Request Format**
+
+        GET /api/completion/v1/stats/<course_key>/
+
+    **Example Requests**
+
+        TODO
+
+    **Response Values**
+
+        TODO
+
+    **Parameters**
+        cohorts (int): TODO Determine if slugs are available too. But how?
+            Specify the cohorts for which to fetch the results.
+            Currently limited to a single cohort, but likely to be expanded
+            later.
+
+        exclude_roles (optional):
+            A comma separated list of roles to exclude from the results.
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid value was sent for requested_fields.
+        * 403 for a user who does not have permission to masquerade as another
+          user specifies a username other than their own.
+        * 404 if the user is not enrolled in the requested course.
+
+        Example response:
+
+            TODO
+
+    """
+
+    def get(self, request, course_key):
+        """
+        Handler for GET requests
+        """
+        course_key = CourseKey.from_string(course_key)
+        paginator = self.pagination_class()  # pylint: disable=not-callable
+        requested_fields = self.get_requested_fields()
+        cohorts = UserCohorts(course_key)
+
+        course_cohorts = cohorts.get_course_cohorts()
+        if not course_cohorts:
+            raise NotFound()
+
+        enrollments = UserEnrollments().get_course_enrollments(course_key)
+        filtered_enrollments = self._filter_enrollments(
+            enrollments=enrollments, course_key=course_key, cohorts=cohorts)
+        paginated = paginator.paginate_queryset(
+            filtered_enrollments, self.request, view=self)
+        aggregator_queryset = self.get_queryset().filter(
+            course_key=course_key,
+            user__in=[enrollment.user for enrollment in paginated])
+        completions = aggregator_queryset.aggregate(
+            possible=Avg('possible'),
+            earned=Avg('earned'),
+            percent=Sum('earned') / Sum('possible'))
+        completions['course_key'] = course_key
+
+        # Return the paginated, serialized completions
+        serializer = self.get_serializer_class()(
+            instance=[completions],
+            requested_fields=requested_fields,
+            many=True
+        )
```

Comment:
This view should define its own serializer.  You asked about this on the ticket, and I'm sorry I wasn't able to respond sooner.  

The other views are effectively returning a list of aggregators for users.  This view is returning a collection of statistics about the course (currently the collection is only a single statistic -- the mean completion, but imagine we add other stats in the future, so the response looks like:

    {
        "results": [{
            "mean_completion": {
                "earned": 45.0, 
                "possible": 100.0, 
                "percent": 0.45 
            },
            "max_completion": {
                "earned": 98.0, 
                "possible": 100.0, 
                "percent": 0.98 
            },
            "most_complete_cohort": 3
        }]
    }

And don't forget to include the envelope information ("course_id", "filters") in the response body.